File: 41-first-missing-positive/first-missing-positive.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        map<int,int> mp;
        for(int i=0;i<nums.size();i++){
            if(nums[i]>0){
                mp[nums[i]]++;
            }
        }
        for(int i=1;i<nums.size()+1;i++){
            mp[i]++;
            if(mp[i]==1){
                return i;
            }
        }
        return nums.size()+1;
    }
};

File: 41-first-missing-positive/first-missing-positive.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        l=set()
        for i in nums:
            if(i>0):
                l.add(i)
        for i in range(1,len(nums)+1):
            if(i not in l):
                return i
        return len(nums)+1
